packages/eas-cli/src/credentials/ios/IosCredentialsProvider.ts

```diff
@@ -36,18 +33,60 @@ export default class IosCredentialsProvider {
   }
 
   private async getLocalAsync(): Promise<IosCredentials> {
-    const credentials = await credentialsJsonReader.readIosCredentialsAsync(this.ctx.projectDir);
-    if (credentialsJsonReader.isCredentialsMap(credentials)) {
-      for (const targetName of Object.keys(credentials)) {
-        this.assertProvisioningProfileType(credentials[targetName].provisioningProfile, targetName);
+    const mainTarget = nullthrows(this.options.targets[0], 'There must be at least one target');
```

Comment:
im not super familiar with the xcode targets, but is it safe to assume that the first thing in the array is always the main target?